gtk: allow running in the background

jcollie opened the pull request:
This patch fixes #2010 by implementing `quit-after-last-window-closed` for the GTK apprt. It also adds the ability for the GTK apprt to exit after a delay once all surfaces have been closed and adds the ability to start Ghostty without opening an initial window.

jcollie committed:
gtk: allow running in the background

This patch fixes #2010 by implementing `quit-after-last-window-closed`
for the GTK apprt. It also adds the ability for the GTK apprt to exit
after a delay once all surfaces have been closed and adds the ability to
start Ghostty without opening an initial window.

mitchellh reviewed (changes requested):
One fairly sizable request but everything else looks good.

jcollie committed:
make quit-after-last-window-closed-delay an optional

jcollie committed:
Improve quit timers.

Instead of "polling" to see if a quit timer has expired, start a single
timer that expires after the confiugred delay when no more surfaces are
open. That timer can be cancelled if necessary.

jparise reviewed (commented):
This looks good to me! Just calling out a few things I noticed while reading through the code.

jcollie committed:
re-use code to cancel old timer

jcollie committed:
make Duration units a struct field to reduce duplication and enable table-driven unit tests

jcollie committed:
fix off-by-one error

jcollie committed:
add a standard zig formatter to Duration and more testing

jcollie committed:
add a warning if quit delay too short

jcollie committed:
document behavior when no initial window is created and a quit delay is set

mitchellh committed:
config: duration prevents overflow, added tests

mitchellh committed:
config: clarify comment

mitchellh committed:
apprt/gtk: use tagged union for quit timer
